### PR TITLE
Project Acceptance Tests

### PR DESCRIPTION
**Pull request by Joel-Milligan**
Added project acceptance tests in a LaTeX document, with the compiled PDF included.

### CODE REVIEW

File: docs/PAT/Project Acceptance Tests.tex [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+\setcounter{tocdepth}{2}
+\setcounter{secnumdepth}{2}
+
+\maketitle
+\tableofcontents
+
+\newpage
+\section{Unit Tests}
+\subsection{Front End Tests}
+All front end JavaScript unit tests, using the \texttt{jest} framework \texttt{react} \texttt{testing} \texttt{library}. Combination of code unit tests and integration test will be used. This makes sure that all components themselves work and the booking system works correctly. Snapshot testing are used as well, to ensure that the DOM and HTML are correct, and any updates that can't be inspected in code shows in the snapshots correctly.
```
Comment (zollf) [inline docs/PAT/Project Acceptance Tests.tex]: `\texttt{react} \texttt{testing} \texttt{library}`
overflows the hbox, so ive just allowed it to break on white spaces. Probably a function for it

File: docs/PAT/Project Acceptance Tests.tex [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+  \item Make changes in the editing page that appears.
+\end{enumerate}
+
+\subsubsection{Expected Result}
+\begin{itemize}
+  \item The changes made should be reflected to normal users who view the car park after the edit.
+\end{itemize}
+
+\subsection{Deleting a Car Park}
+An admin use should be able to delete existing car parks.
```
Comment (APM246) [inline docs/PAT/Project Acceptance Tests.tex]: spelling

File: docs/PAT/Project Acceptance Tests.tex (hunk 1 above)
Comment (APM246) [inline docs/PAT/Project Acceptance Tests.tex]: Did you mean to say "All front end JavaScript unit tests, using the jest framework react testing library , *should pass*"

File: docs/PAT/Project Acceptance Tests.tex [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+\end{enumerate}
+
+\subsubsection{Expected Results}
+\begin{itemize}
+  \item All unit tests pass.
+  \item At least 90\% test coverage.
+\end{itemize}
+
+\subsection{Back End Tests}
+All back end Python unit tests, using django's built in testing library, should pass. Combination of code unit tests and integration test will be used. This includes testing endpoints, models and serializers to make sure they work as expected. Backend will use \texttt{sqlite3} as a dummy database to test on for creating, reading, inserting, and deleting data. This is to make it easier to code integration, rather than hosting a \texttt{mysql} database test server.
```
Comment (APM246) [inline docs/PAT/Project Acceptance Tests.tex]: "as a dummy database to test creating, reading, inserting and deleting data"

Comment (APM246) [inline docs/PAT/Project Acceptance Tests.tex]: "to make it easier for code integration"

File: docs/PAT/Project Acceptance Tests.tex [hunk 4]
```diff
@@ -0,0 +1,245 @@
...
+
+\subsubsection{Expected Results}
+\begin{itemize}
+  \item All unit tests pass.
+  \item At least 75\% test coverage.
+\end{itemize}
+
+\newpage
+\section{Code Linting}
+Code linting will be used in both frontend and backend to make sure code formatting is readable, up to a standard and the check on little human errors. For frontend we use a combination of \texttt{prettier} and \texttt{eslint}, and for backend \texttt{pycodestyle}. These are industry standards for linting.
```
Comment (APM246) [inline docs/PAT/Project Acceptance Tests.tex]: Maybe "Code linting will be used in both frontend and backend to check for errors and ensure consistent formatting" sounds better

File: docs/PAT/Project Acceptance Tests.tex [hunk 5]
```diff
@@ -0,0 +1,245 @@
...
+  \item Run the command \texttt{pycodestyle app/}
+\end{enumerate}
+
+\subsubsection{Expected Results}
+\begin{itemize}
+  \item No linting errors.
+\end{itemize}
+
+\section{CI/CD}
+Continuous integration and deployment with GitHub action is an automatic way do testing and make sure the product can build correctly. Every change to the code base will need to be reviewed, and be put into a pipeline to check all code test and linting.
```
Comment (APM246) [inline docs/PAT/Project Acceptance Tests.tex]: code tests instead of code test

File: docs/PAT/Project Acceptance Tests.tex [hunk 6]
```diff
@@ -68,11 +68,11 @@ \section{CI/CD}
 \newpage
 \section{Creating Bookings}
 \subsection{Accessible Website}
-The booking part of the website should be accessible to anyone with a modern web browser.
+The booking part of the website should be accessible to anyone with a modern web browser and the two main operating system (Mac and Windows).
```
Comment (APM246) [inline docs/PAT/Project Acceptance Tests.tex]: systems instead of system

**Review by APM246 — APPROVED** [review]

**Pull request merged**